src/setup/stateful.rs

```diff
@@ -0,0 +1,72 @@
+//! Contains code to start and stop a node with preloaded ledger data.
+
+use std::{
+    net::{Ipv4Addr, SocketAddr, SocketAddrV4},
+    path::PathBuf,
+    process::Child,
+};
+
+use crate::setup::{config::NodeMetaData, process, process::wait_for_start};
+
+const STATEFUL_PORT: u16 = 5005;
+
+/// Stateful node. Binds to 127.0.0.1:5005, loads data and configuration from `~/.ziggurat/ripple_stateful`.
+/// Does *not* clean its database after stopping.
+pub struct StatefulNode {
+    /// The process encapsulating the running node.
+    process: Option<Child>,
+}
+
+impl StatefulNode {
+    /// Build and starts a new stateful node.
+    pub async fn new(config_path: PathBuf, log_to_stdout: bool) -> anyhow::Result<Self> {
+        let mut meta = NodeMetaData::new(config_path)?;
+        meta.start_args.append(&mut vec![
```

Comment:
yeah, we don't want a separate `Node` setup; this should just optionally be done in the plain `Node`

Reply:
Moved to separate function in Node.